tests/voq/test_voq_ipfwd.py

```diff
@@ -728,45 +744,70 @@ def test_voq_dut_lb_ping(self, duthosts, all_cfg_facts, ttl, size, version, port
+            check_packet(sonic_ping, ports, 'portC', 'portA', src_ip_fld=my_src_fld, dst_ip_fld='my_lb_ip', size=size,
                          ttl=ttl, ttl_change=0)
         else:
-            check_packet(sonic_ping, ports, 'portC', 'portA', src_ip_fld='my_lb_ip', dst_ip_fld='my_lb_ip', size=size,
+            if ports['portC']['dut'].get_facts()['num_asic'] > 1:
+                # Remote asic is part of a multi-asic linecard, then all asics of the remote card have the same
+                # loopback, so we can't be sure if we are picking the right asic to ping - so use Loopback4096 address.
+                check_packet(sonic_ping, ports, 'portC', 'portA', src_ip_fld=my_src_fld, dst_ip_fld='my_lb4096_ip',
+                             size=size, ttl=ttl, ttl_change=1)
+            else:
+                check_packet(sonic_ping, ports, 'portC', 'portA', src_ip_fld=my_src_fld, dst_ip_fld='my_lb_ip',
+                             size=size, ttl=ttl, ttl_change=1)
+
+        if ports['portA']['dut'].get_facts()['num_asic'] == 1:
+            # Remote nbr on remote asic will only have Loopback0 address which is unique only for single asic.
+            # For multi-asic linecard, all asics, would have the same Loopback0, but different Loopback4096. But,
+            # Loopback4096 is not advertised to the nbrs.
+            check_packet(sonic_ping, ports, 'portC', 'portA', src_ip_fld=my_src_fld, dst_ip_fld='nbr_ip', size=size,
+                         ttl=ttl, ttl_change=1)
+            check_packet(sonic_ping, ports, 'portC', 'portA', src_ip_fld=my_src_fld, dst_ip_fld='nbr_lb', size=size,
                          ttl=ttl, ttl_change=1)
-
-        check_packet(sonic_ping, ports, 'portC', 'portA', src_ip_fld='my_lb_ip', dst_ip_fld='nbr_ip', size=size,
-                     ttl=ttl, ttl_change=1)
-        check_packet(sonic_ping, ports, 'portC', 'portA', src_ip_fld='my_lb_ip', dst_ip_fld='nbr_lb', size=size,
-                     ttl=ttl, ttl_change=1)
         if ports['portD']['dut'].get_facts()['asic_type'] == 'vs':
-            check_packet(sonic_ping, ports, 'portD', 'portA', src_ip_fld='my_lb_ip', dst_ip_fld='my_lb_ip', size=size, ttl=ttl,
+            check_packet(sonic_ping, ports, 'portD', 'portA', src_ip_fld=my_src_fld, dst_ip_fld='my_lb_ip', size=size, ttl=ttl,
                          ttl_change=0)
         else:
-            check_packet(sonic_ping, ports, 'portD', 'portA', src_ip_fld='my_lb_ip', dst_ip_fld='my_lb_ip', size=size, ttl=ttl,
-                         ttl_change=1)
-        check_packet(sonic_ping, ports, 'portD', 'portA', src_ip_fld='my_lb_ip', dst_ip_fld='nbr_ip', size=size,
-                     ttl=ttl, ttl_change=1)
-        check_packet(sonic_ping, ports, 'portD', 'portA', src_ip_fld='my_lb_ip', dst_ip_fld='nbr_lb', size=size,
-                     ttl=ttl, ttl_change=1)
+            if ports['portD']['dut'].get_facts()['num_asic'] > 1:
```

Comment:
We could do tht ping to Loopback0 for single and multi asic card, in addition to that for multi asic we ping loopback 4096 ?

Reply:
Also I think in the recent multi asic minigraph changes, the Loopback 4096 will also added for single asic line cards

Reply:
It looks like for single ASIC, Loopback4096 is in HIDE_INTERNAL group in the BGP config, so it can't be pinged.